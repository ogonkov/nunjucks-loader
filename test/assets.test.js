/* globals __USE_ES__ */

import compiler from './compiler';

const loaderOptions = {
    esModule: __USE_ES__,
    assetsPaths: [
        'test/fixtures/django_project/app_example/static'
    ]
};

describe('Assets', function() {
    test('should load static assets', async function() {
        const output = await compiler('fixtures/assets/template.njk', loaderOptions);

        await expect(output()).resolves.toMatchSnapshot();
    });

    test('should load exported static assets', async function() {
        const output = await compiler('fixtures/assets/template-as.njk', loaderOptions);

        await expect(output()).resolves.toMatchSnapshot();
    });

    test('should load dynamic assets', async function() {
        const output = await compiler('fixtures/assets/dynamic.njk', loaderOptions);

        await expect(output()).resolves.toMatchSnapshot();
    });

    test('should load dynamic assets with trailing slashes', async function() {
        const output = await compiler('fixtures/assets/dynamic-trailing-slash.njk', loaderOptions);

        await expect(output()).resolves.toMatchSnapshot();
    });

    test('should load exported dynamic assets', async function() {
        const output = await compiler('fixtures/assets/dynamic-as.njk', loaderOptions);

        await expect(output()).resolves.toMatchSnapshot();
    });

    it('should load variables', async function() {
        const output = await compiler('fixtures/assets/dynamic-variable.njk', loaderOptions);

        await expect(output()).resolves.toMatchSnapshot();
    });
});
