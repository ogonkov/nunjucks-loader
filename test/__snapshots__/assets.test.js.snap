// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Assets should load dynamic assets 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <meta charset=\\"utf-8\\">
    <title></title>
    <link rel=\\"stylesheet\\" href=\\"bundles/fixtures/django_project/app_example/static/foo/bar/style.css\\"/>
  </head>
  <body>
    
    

    
        <p>bundles/fixtures/django_project/app_example/static/foo/bar/dynamic-example-1.md</p>
    
        <p>bundles/fixtures/django_project/app_example/static/foo/bar/dynamic-example-2.md</p>
    

    
        <p>bundles/fixtures/django_project/app_example/static/foo/bar/dynamic-example-1.md</p>
    
        <p>bundles/fixtures/django_project/app_example/static/foo/bar/dynamic-example-2.md</p>
    

  </body>
</html>
"
`;

exports[`Assets should load dynamic assets with trailing slashes 1`] = `
"
<p>bundles/fixtures/django_project/app_example/static/foo/bar/dynamic-example-1.md</p>

<p>bundles/fixtures/django_project/app_example/static/foo/bar/dynamic-example-2.md</p>

"
`;

exports[`Assets should load exported dynamic assets 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <meta charset=\\"utf-8\\">
    <title></title>
    <link rel=\\"stylesheet\\" href=\\"bundles/fixtures/django_project/app_example/static/foo/bar/style.css\\"/>
  </head>
  <body>
    
    

    
        <p>bundles/fixtures/django_project/app_example/static/foo/bar/dynamic-example-1.md</p>
    
        <p>bundles/fixtures/django_project/app_example/static/foo/bar/dynamic-example-2.md</p>
    

    
        <p>bundles/fixtures/django_project/app_example/static/foo/bar/dynamic-example-1.md</p>
    
        <p>bundles/fixtures/django_project/app_example/static/foo/bar/dynamic-example-2.md</p>
    

  </body>
</html>
"
`;

exports[`Assets should load exported static assets 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <meta charset=\\"utf-8\\">
    <title></title>
    <link rel=\\"stylesheet\\" href=\\"bundles/fixtures/django_project/app_example/static/foo/bar/style.css\\"/>
  </head>
  <body>
    
    <p><a href=\\"bundles/fixtures/django_project/app_example/static/foo/bar/test.txt\\">Click here</a></p>

  </body>
</html>
"
`;

exports[`Assets should load static assets 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <meta charset=\\"utf-8\\">
    <title></title>
    <link rel=\\"stylesheet\\" href=\\"bundles/fixtures/django_project/app_example/static/foo/bar/style.css\\"/>
  </head>
  <body>
    
    <p><a href=\\"bundles/fixtures/django_project/app_example/static/foo/bar/test.txt\\">Click here</a></p>

  </body>
</html>
"
`;

exports[`Assets should load variables 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <meta charset=\\"utf-8\\">
    <title></title>
    <link rel=\\"stylesheet\\" href=\\"bundles/fixtures/django_project/app_example/static/foo/bar/style.css\\"/>
  </head>
  <body>
    
    
        <p>bundles/fixtures/django_project/app_example/static/foo/bar/dynamic-example-1.md</p>
    
        <p>bundles/fixtures/django_project/app_example/static/foo/bar/dynamic-example-2.md</p>
    

  </body>
</html>
"
`;
